src/utils/build-url.ts

```diff
@@ -1,12 +1,22 @@
 import { URL } from 'url';
+import { stringify } from 'querystring';
 
 export function buildUrl(
```

Comment:
this function became a bit fuzzy - it returns url to copletely different endpoints depending on whenever you passed an optional arguments or not. 

We should rather split this into two separate functions, or even inline this logic into respective modules. This functionality  is so small that it may not be worth extracting to standalone module